branchLayout/api/src/main/java/com/virtuslab/branchlayout/api/BranchLayout.java

```diff
@@ -29,6 +30,28 @@ private static List<IBranchLayoutEntry> collectEntriesRecursively(IBranchLayoutE
     return entryByName.get(branchName).getOrNull();
   }
 
+  @Override
+  public @Nullable IBranchLayoutEntry findNextEntry(String branchName) {
+    val entriesOrderedList = rootEntries.flatMap(BranchLayout::collectEntriesRecursively).map(IBranchLayoutEntry::getName);
+    val currentIndex = entriesOrderedList.indexOf(branchName);
+    if (currentIndex > -1 && currentIndex + 1 < entriesOrderedList.length()) {
+      @LTLengthOf("entriesOrderedList") int nextIndex = currentIndex + 1;
+      return findEntryByName(entriesOrderedList.get(nextIndex));
+    }
+    return null;
+  }
+
+  @Override
+  public @Nullable IBranchLayoutEntry findPreviousEntry(String branchName) {
+    val entriesOrderedList = rootEntries.flatMap(BranchLayout::collectEntriesRecursively).map(IBranchLayoutEntry::getName);
+    val currentIndex = entriesOrderedList.indexOf(branchName);
+    if (currentIndex > 0 && currentIndex < /* added for preventing a type error */entriesOrderedList.length() - 1) {
```

Comment:
```suggestion
    if (currentIndex > 0 && currentIndex < /* added for preventing a type error */entriesOrderedList.length() + 1) {
```

Reply:
because we want `currentIndex - 1 < length`

Reply:
I.e. `currentIndex <= entriesOrderedList.length()`? 🤔 is it even possible to get an index like that from `indexOf`?

Reply:
>  is it even possible to get an index equal to entriesOrderedList.length() from indexOf
oh, our checker cannot deduct that. But I removed the `-1` which was causing the null return.